wgpu-core/src/lock/mod.rs

```diff
@@ -0,0 +1,22 @@
+/*! Deadlock prevention via lock ordering.
+
+This module defines wrappers for the lock types used in `wgpu`
+(`Mutex`, `RwLock`, and `SnatchLock`) that use runtime checks to
+ensure that locks are always acquired in a consistent order, ensuring
+that deadlock is impossible. These checks are enabled only in debug
+builds, and when the `"validate-locks"` feature is enabled.
+
+See the [`validating`] module for details on how/why the validation works.
+
+*/
```

Comment:
I guess I wished I'd used it for the `wgpu-hal` module docs, because it would have made it easier to paste into the `README.md` file. (Yes, I know there is tooling for that.)

I've changed this to use `\\!`.